### PR TITLE
chart-testing 3.0.0 (new formula)

### PR DESCRIPTION
**Pull request by crunchtime-ali**
This formula lints and validates Helm charts.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/chart-testing.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  depends_on "go" => :build
+
+  def install
+    commit = Utils.safe_popen_read("git", "rev-parse", "HEAD").chomp
+    ldflags = %W[
+      -X github.com/helm/chart-testing/v3/ct/cmd.Version=#{version}
+      -X github.com/helm/chart-testing/v3/ct/cmd.GitCommit=#{commit}
+      -X github.com/helm/chart-testing/v3/ct/cmd.BuildDate=#{Date.today}
+    ]
+    system "go", "build", "-ldflags", ldflags.join(" "), "-trimpath", "-o", bin/"ct", "./ct/main.go"
```
Comment (dawidd6) [inline Formula/chart-testing.rb:18]:
```suggestion
    system "go", "build", *std_go_args, "-ldflags", ldflags, "-o", bin/"ct", "./ct/main.go"
```

File: Formula/chart-testing.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  depends_on "go" => :build
+
+  def install
+    commit = Utils.safe_popen_read("git", "rev-parse", "HEAD").chomp
+    ldflags = %W[
+      -X github.com/helm/chart-testing/v3/ct/cmd.Version=#{version}
+      -X github.com/helm/chart-testing/v3/ct/cmd.GitCommit=#{commit}
+      -X github.com/helm/chart-testing/v3/ct/cmd.BuildDate=#{Date.today}
+    ]
```
Comment (dawidd6) [inline Formula/chart-testing.rb:17]:
```suggestion
    ].join(" ")
```

File: Formula/chart-testing.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    system "go", "build", "-ldflags", ldflags.join(" "), "-trimpath", "-o", bin/"ct", "./ct/main.go"
+  end
+
+  test do
+    assert_match "Lint and test", shell_output("#{bin}/ct --help")
+
+    # Tries to lint the Helm charts in the current directory which fails
+    # because there are no Helm charts in the current directory
```
Comment (SMillerDev) [inline Formula/chart-testing.rb:25]: Those aren't very complex. Why not just write a very basic chart to lint?

Comment (crunchtime-ali) [reply Formula/chart-testing.rb]: Now I just create a new Helm chart that will be used for testing with `helm create testchart`

File: Formula/chart-testing.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    system "go", "build", *std_go_args, "-ldflags", ldflags, "-o", bin/"ct", "./ct/main.go"
+  end
+
+  test do
+    assert_match "Lint and test", shell_output("#{bin}/ct --help")
+
+    # Lint an empty Helm chart that we create with `helm create`
+    system "helm", "create", "testchart"
+    output = shell_output("#{bin}/ct lint --charts ./testchart --validate-chart-schema=false" \
+                          "--validate-maintainers=false").lines.last.chomp
```
Comment (SMillerDev) [inline Formula/chart-testing.rb:29]: It is concatenating the flags
```suggestion
                          " --validate-maintainers=false").lines.last.chomp
```

**Comment by crunchtime-ali** [comment]
Any ideas why `brew test-bot --only-tap-syntax` errors with:

```
Error: 1 failed steps!
brew style homebrew/core
```

**Comment by dawidd6** [comment]
Will be fixed soon, don't worry.

**Comment by dawidd6** [comment]
You may need to rebase.

**Comment by crunchtime-ali** [comment]
> You may need to rebase.

And change to the Ruby 1.9 hash syntax ;)
Done

File: Formula/chart-testing.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  url "https://github.com/helm/chart-testing.git",
+      {
+        tag:      "v3.0.0",
+        revision: "50db473a1e68c605b18d82f019d83ea401542213",
+      }
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+  depends_on "helm" => :test
+  depends_on "yamllint" => :yamllint
```
Comment (dawidd6) [inline Formula/chart-testing.rb:13]: What `:yamlint` means here?

File: Formula/chart-testing.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
+class ChartTesting < Formula
+  desc "Testing and linting Helm charts"
+  homepage "https://github.com/helm/chart-testing"
+  url "https://github.com/helm/chart-testing.git",
+      {
+        tag:      "v3.0.0",
+        revision: "50db473a1e68c605b18d82f019d83ea401542213",
+      }
```
Comment (dawidd6) [inline Formula/chart-testing.rb:8]: @Homebrew/core is this correct with the new hash syntax?

Comment (SeekingMeaning) [inline Formula/chart-testing.rb:8]:
```suggestion
      tag:      "v3.0.0",
      revision: "50db473a1e68c605b18d82f019d83ea401542213"
```

File: Formula/chart-testing.rb (hunk 5 above)
Comment (crunchtime-ali) [reply Formula/chart-testing.rb]: This was meant to be `:test`. I changed it now.

File: Formula/chart-testing.rb (hunk 6 above)
Comment (crunchtime-ali) [reply Formula/chart-testing.rb]: @SeekingMeaning Yes it does but I changed it according to @SeekingMeaning's idea.

**Review by dawidd6 — APPROVED** [review]

**Comment by sc250024** [comment]
Nice, thank you for doing this @crunchtime-ali

**Pull request closed without merging**